### PR TITLE
Feat/material icon/custom tags

### PR DESCRIPTION
**Pull request by mgr34**
#689 

---
I signed it

### CODE REVIEW

File: packages/material-icon/README.md [hunk 1]
```diff
@@ -52,10 +52,42 @@ const MyComponent = () => {
...
+tag | String | Customizes the MaterialIcon HTML tag. (default: `<i>`) 
+
 
 ## Usage with Icons
 
 Please see our [Best Practices doc](../../docs/best-practices.md#importing-font-icons) for further reading.
+
+## Usage with `tag='button'` 
+
+if you set the `MaterialIcon` property tag to button you may find it desirable
```
Comment (moog16) [inline packages/material-icon/README.md]: If*

File: packages/material-icon/index.tsx [hunk 2]
```diff
@@ -60,23 +70,26 @@ export default class MaterialIcon extends React.Component<
     className: '',
     hasRipple: false,
     unbounded: false,
+    tag: 'i',
   };
 
   render() {
-    const {icon, hasRipple, unbounded, ...otherProps} = this.props; // eslint-disable-line no-unused-vars
+    const {icon, hasRipple, unbounded, tag, ...otherProps} = this.props; // eslint-disable-line no-unused-vars
```
Comment (moog16) [inline packages/material-icon/index.tsx]: This actually doesn't need to be here (nor does icon).

File: packages/material-icon/index.tsx [hunk 3]
```diff
@@ -60,23 +70,26 @@ export default class MaterialIcon extends React.Component<
...
     if (hasRipple) {
       return (
         <RippleMaterialIcon
           unbounded
           hasRipple
           icon={icon}
+          tag={tag}
```
Comment (moog16) [inline packages/material-icon/index.tsx]: remove this and icon.

File: packages/material-icon/index.tsx [hunk 4]
```diff
@@ -60,23 +70,26 @@ export default class MaterialIcon extends React.Component<
...
           {...otherProps}
         />
       );
     }
     return (
       <MaterialIconDefault
         icon={icon}
+        tag={tag}
```
Comment (moog16) [inline packages/material-icon/index.tsx]: remove this and icon.

File: test/unit/material-icon/index.test.tsx [hunk 5]
```diff
@@ -26,15 +36,26 @@ test('has icon type as child', () => {
...
+const rippledTaggedIconComponent = (
+  <RippleMaterialIcon hasRipple unbounded icon='menu' tag='a' />
 );
 
 test('if hasRipple true, then it should contain RippleMaterialIcon', () => {
   const wrapper = mount(<MaterialIcon icon='menu' hasRipple />);
   assert.isTrue(wrapper.contains(rippledIconComponent));
 });
 
+test('if hasRipple true prop.tag, then it should contain RippleMaterialIcon with custom tag', () => {
```
Comment (moog16) [inline test/unit/material-icon/index.test.tsx]: can you add a test that adds an href? I think the typescript compiler might error.

Comment (mgr34) [reply test/unit/material-icon/index.test.tsx]: Oh yes, it does throw an error. I've been a bit busy with other work the past week. I'm probably going have to delay this PR for a bit. I'll try to get it set for the next release. Do you have a date in mind for that?

Comment (moog16) [reply test/unit/material-icon/index.test.tsx]: Sounds good! Milestone 12 is going to happen April 16th.

**Comment by moog16** [comment]
@mgr34 - were you going to continue to work on this? I think if you're not going to get to it within the next 2 weeks, we should close it.

**Comment by mgr34** [comment]
Hi @moog16 - I did some work a few weeks back that was never reviewed. (#571e10b) Looking at it now I see an errant search and replace has messed up the license at the top . I could fix that if you have time to review the rest of the changes made.  Otherwise, I do not have much  time available in the next two weeks. Other projects will be keeping me busy until the 2nd week of may.

**Comment by mgr34** [comment]
I signed it

**Comment by moog16** [comment]
@mgr34 - we can keep it open. I just wanted to see if you were going to work on it again. We can wait until next release to start reviewing. Once you've finish the changes you need to make, please let me know and i'll review again.

**Comment by moog16** [comment]
Closing due to inactivity. We can pick this up again when you have time :)

**Pull request closed without merging**